runtime/common/logging_xwalk.cc

```diff
@@ -0,0 +1,204 @@
+// Copyright (c) 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "build/build_config.h"
+
+// Need to include this before most other files because it defines
+// IPC_MESSAGE_LOG_ENABLED. We need to use it to define
+// IPC_MESSAGE_MACROS_LOG_ENABLED so render_messages.h will generate the
+// ViewMsgLog et al. functions.
+#include "ipc/ipc_message.h"
+
+// On Windows, the about:ipc dialog shows IPCs; on POSIX, we hook up a
+// logger in this file.  (We implement about:ipc on Mac but implement
+// the loggers here anyway).  We need to do this real early to be sure
+// IPC_MESSAGE_MACROS_LOG_ENABLED doesn't get undefined.
+#if defined(OS_POSIX) && defined(IPC_MESSAGE_LOG_ENABLED)
+#define IPC_MESSAGE_MACROS_LOG_ENABLED
+#include "content/public/common/content_ipc_logging.h"
+#define IPC_LOG_TABLE_ADD_ENTRY(msg_id, logger) \
+    content::RegisterIPCLogger(msg_id, logger)
+#endif
+
+#if defined(OS_WIN)
+#include <windows.h>
+#endif
+
+#include "xwalk/runtime/common/logging_xwalk.h"
+
+#include <fstream>  // NOLINT
+#include <string>  // NOLINT
+
+#include "base/command_line.h"
+#include "base/compiler_specific.h"
+#include "base/debug/debugger.h"
+#include "base/debug/dump_without_crashing.h"
+#include "base/files/file_path.h"
+#include "base/files/file_util.h"
+#include "base/path_service.h"
+#include "base/strings/string_number_conversions.h"
+#include "base/strings/string_util.h"
+#include "base/strings/stringprintf.h"
+#include "base/strings/utf_string_conversions.h"
+#include "base/base_switches.h"
+#include "base/threading/thread_restrictions.h"
+#include "content/public/common/content_switches.h"
+#include "ipc/ipc_logging.h"
+#include "xwalk/runtime/common/xwalk_paths.h"
+
+#if defined(OS_WIN)
+#include <initguid.h>
+#include "base/logging_win.h"
+#endif
+
+namespace {
+
+// This should be true for exactly the period between the end of
+// InitXwalkLogging() and the beginning of CleanupXwalkLogging().
+bool xwalk_logging_initialized_ = false;
+
+// Set if we called InitXwalkLogging() but failed to initialize.
+bool xwalk_logging_failed_ = false;
+
+// This should be true for exactly the period between the end of
+// InitXwalkLogging() and the beginning of CleanupXwalkLogging().
+bool xwalk_logging_redirected_ = false;
+
+#if defined(OS_WIN)
+// {7FE69228-633E-4f06-80C1-527FEA23E3A7}
+const GUID kXwalkTraceProviderName = {
+    0x7fe69228, 0x633e, 0x4f06,
+        { 0x80, 0xc1, 0x52, 0x7f, 0xea, 0x23, 0xe3, 0xa7 } };
```

Comment:
I checked that if we simultaneously enable chrome and xwalk logging, there won't be any conflict. So the concern is not necessary.